lib/modules/dosomething/dosomething_signup/includes/Signup.php

```diff
@@ -92,5 +90,6 @@ private function build($data) {
     $this->created_at = $data->timestamp;
     $this->campaign = Campaign::get($data->nid);
     $this->campaign_run = $data->run_nid;
+    $this->rbid = Reportback::get($data->rbid);
```

Comment:
You're right on line 93 causing an issue. There seems to be a problem with some of the staging data on some super early reportbacks (like rbid 2 & 3 for example) where while there's an RB there are absolutely no RB Items in the db for it and it totally :hankey:s out :scream: However, since the `Reportback::get()` throws an exception we can catch it and deal with it. So I tried something like this, and the index collection started working correctly:

```
private function build($data) {
    $this->id = $data->sid;
    $this->created_at = $data->timestamp;
    $this->campaign = Campaign::get($data->nid);
    $this->campaign_run = $data->run_nid;

    try {
      $this->rbid = Reportback::get($data->rbid);
    }
    catch (Exception $error) {
      $this->rbid = null;
    }
  }
```